Add API to create stub SIP objects; allow SIPs to be started from stubs

mistydemeo opened the pull request:
- Adds a new API to create an unstarted SIP object. A SIP is understood to be "unstarted" if its `currentpath` is null.
- Allows the SIP arrange "create SIP" API to accept a SIP UUID. If that UUID points to an unstarted SIP, uses that as the SIP object and updates it with the correct path. If it points to a started SIP, the API returns 400. If it doesn't point to a SIP that exists, then a SIP object is created with that UUID.

mistydemeo committed:
Add API to create stub SIP and retrieve UUID

mistydemeo committed:
Allow creating a SIP using an existing SIP object

Fixes #9410.

Hwesta commented on src/dashboard/src/components/filesystem_ajax/views.py:
```diff
@@ -331,8 +331,19 @@ def create_arranged_sip(staging_sip_path, files, sip_uuid):
+        sip.save()
+    # Create a SIP object if none exists
+    else:
+        databaseFunctions.createSIP(currentpath, sip_uuid)
```
This seems like it would be easier with a `.get` and try/catch? eg

```
try:
    sip = models.SIP.objects.get(uuid=sip_uuid)
    if sip.currentpath is not None:
        return "Provided SIP UUID ({}) belongs to an already-started SIP!".format(sip_uuid)
    sip.currentpath = currentpath
    sip.save()
except models.SIP.DNE:
    databaseFunctions.createSIP(currentpath, sip_uuid)
```

mistydemeo replied:
Updated, though I moved most of the block into an `else`.

Hwesta commented:
:+1:

Hwesta commented:
:+1:

mistydemeo committed:
Add access URLs which redirect to metadata, rights

These create a stub SIP object if necessary.

mistydemeo committed:
Access.start_sip: use existing SIP, if created

mistydemeo commented:
Made a few tweaks and added some new commits:
- Added access URLs which redirect to metadata and rights, creating a stub SIP if necessary.
- Access's `start_sip` will look up a proto-SIP if one exists, and pass that to the underlying view.

Hwesta commented:
:+1:
